icu4c/source/common/lstmbe.cpp

```diff
@@ -589,36 +589,29 @@ void GraphemeClusterVectorizer::vectorize(
 
 // Computing LSTM as stated in
 // https://en.wikipedia.org/wiki/Long_short-term_memory#LSTM_with_a_forget_gate
+// ifco, hU and ic are temp array allocate outside which do not need to be
+// input/output value but could avoid unnecessary memory alloc/free if passing
+// in.
 void compute(
+    int32_t hunits,
     const ReadArray2D& W, const ReadArray2D& U, const ReadArray1D& b,
     const ReadArray1D& x, Array1D& h, Array1D& c,
-    UErrorCode &status)
+    Array1D& ifco, Array1D& hU, Array1D& ic)
 {
-    if (U_FAILURE(status)) return;
     // ifco = x * W + h * U + b
-    Array1D ifco(b.d1(), status);
-    {
-        Array1D hU(b.d1(), status);
-        if (U_FAILURE(status)) return;
-        ifco.dotProduct(x, W)
-            .add(hU.dotProduct(h, U))
-            .add(b);
-    } // delocate hU
+    ifco.dotProduct(x, W)
+        .add(hU.dotProduct(h, U))
```

Comment:
Suggestion (optional): Add a new specialized function `.addDotProduct` that avoids the need for allocating `hU`